py-polars/tests/unit/test_show_versions.py

```diff
@@ -8,3 +8,12 @@ def test_show_versions(capsys) -> None:  # type: ignore[no-untyped-def]
     assert "Python" in out
     assert "Polars" in out
     assert "Optional dependencies" in out
+
+
+def test_build_info() -> None:
+    build_info = pl.build_info()
+    features = build_info.get("features", {})
+    if features:
```

Comment:
It would be nice if we can test both branches here somehow.

Reply:
I think the code already tests both branches:

1. When `build_info` feature gate in enabled `features` field is populated and tested for presence of `BUILD_INFO` feature
2. Otherwise without `build_info` feature the only element present in dictionary is `version` and its presence is tested in the `else` part of the code (line 17, not shown in the snippet).

Reply:
Moved testing code into `test_build_info.py` file.